nix/k8s-tests/src/k8s_tests_tester.py

```
@@ -1200,6 +1214,426 @@ def _test_s3_storage(self, deployment_config: dict) -> TestResult:
+        credential env vars all match the serving deployment exactly).
+        Returns combined stdout+stderr.
+        """
+        namespace = deployment_config["namespace"]
+        pods = self.k8s_core_v1.list_namespaced_pod(
+            namespace=namespace, label_selector="app.kubernetes.io/name=ncps"
+        )
+        running = [p for p in pods.items if p.status.phase == "Running"]
+        if not running:
+            raise RuntimeError("no running ncps pod to exec migrate-chunks-to-nar")
+        pod_name = running[0].metadata.name
+        result = subprocess.run(
```

Ruff (0.15.15)

[error] `subprocess` call: check for execution of untrusted input

(S603)